src/clusterfuzz/_internal/bot/tasks/utasks/minimize_task.py

```diff
@@ -555,52 +563,115 @@ def utask_main(uworker_input):
+
+  minimize_task_output.last_crash_result_dict.clear()
+  minimize_task_output.last_crash_result_dict.update(
+      _extract_crash_result(last_crash_result, command))
 
-  minimize_task_output = uworker_msg_pb2.MinimizeTaskOutput(
-      last_crash_result_dict=_extract_crash_result(last_crash_result, command),
-      flaky_stack=flaky_stack)
   return uworker_msg_pb2.Output(minimize_task_output=minimize_task_output)
 
 
 HANDLED_ERRORS = [
     uworker_msg_pb2.ErrorType.MINIMIZE_SETUP,
     uworker_msg_pb2.ErrorType.TESTCASE_SETUP,
+    uworker_msg_pb2.ErrorType.MINIMIZE_UNREPRODUCIBLE_CRASH,
+    uworker_msg_pb2.ErrorType.MINIMIZE_CRASH_TOO_FLAKY,
+    uworker_msg_pb2.ErrorType.MINIMIZE_DEADLINE_EXCEEDED,
+    uworker_msg_pb2.ErrorType.MINIMIZE_DEADLINE_EXCEEDED_IN_MAIN_FILE_PHASE,
+    uworker_msg_pb2.ErrorType.LIBFUZZER_MINIMIZATION_UNREPRODUCIBLE,
+    uworker_msg_pb2.ErrorType.LIBFUZZER_MINIMIZATION_FAILED,
     uworker_msg_pb2.ErrorType.UNHANDLED
 ]
 
 
+def update_testcase(output: uworker_msg_pb2.Output):
+  """Updates the tescase using the values passed from utask_main. This done at
+  the beginning of utask_postprocess and before error handling is called."""
+  if not output.HasField('minimize_task_output'):
+    return
+
+  minimize_task_output = output.minimize_task_output
+  testcase = data_handler.get_testcase_by_id(output.uworker_input.testcase_id)
+
+  _update_testcase_memory_tool_options(testcase,
+                                       minimize_task_output.memory_tool_options)
+
+  if minimize_task_output.security_severity_updated:
+    if minimize_task_output.HasField('security_severity'):
+      testcase.security_severity = minimize_task_output.security_severity
+    else:
+      testcase.security_severity = None
+
+  if minimize_task_output.HasField('minimization_phase'):
+    testcase.set_metadata('minimization_phase',
+                          minimize_task_output.minimization_phase)
+
+  if minimize_task_output.flaky_stack:
+    testcase.flaky_stack = minimize_task_output.flaky_stack
+
+  if minimize_task_output.HasField('minimized_arguments'):
+    testcase.minimized_arguments = minimize_task_output.minimized_arguments
+
+  if minimize_task_output.HasField('archive_state'):
+    testcase.archive_state = minimize_task_output.archive_state
+
+  if minimize_task_output.HasField('absolute_path'):
+    testcase.absolute_path = minimize_task_output.absolute_path
+
+  if minimize_task_output.gestures:
+    testcase.gestures = minimize_task_output.gestures
+
+  if minimize_task_output.HasField('minimized_keys'):
+    testcase.minimized_keys = minimize_task_output.minimized_keys
+
+  testcase.put()
+
+
 def utask_postprocess(output):
   """Postprocess in a trusted bot."""
+  update_testcase(output)
   if output.error_type != uworker_msg_pb2.ErrorType.NO_ERROR:
     uworker_handle_errors.handle(output, HANDLED_ERRORS)
     return
 
+  if environment.is_engine_fuzzer_job():
```

Comment:
Unsure if this condition is correct here. 

The previous implementation only reaches this if it's an engine **AND** not libfuzzer (because there is a previous check for libFuzzer). 

We could also just move this to pre-process, and skip this step completely in such cases (i.e. a non libfuzzer engine case).

Reply:
Indeed thanks for catching that! I moved the check and the handling to preprocess.